reconcile/utils/terrascript_client.py

```diff
@@ -3651,64 +3637,65 @@ def populate_tf_resource_alb(self, resource, namespace_info,
-        lblr_read_tf_resource = \
-            aws_lb_listener_rule(lblr_read_identifier, **values)
-        tf_resources.append(lblr_read_tf_resource)
-        # write
-        write_weights = [t['weight'] for t in write_weighted_target_groups]
-        if sum(write_weights) != 100:
-            raise ValueError('sum of weights of targets should be 100')
-        values = {
-            'listener_arn': f'${{{forward_lbl_tf_resource.arn}}}',
-            'action': {
-                'type': 'forward',
-                'forward': {
-                    'target_group': write_weighted_target_groups,
-                    'stickiness': {
-                        'enabled': False,
-                        'duration': 1,  # required
-                    },
+
+        for rule_num, rule in enumerate(resource['rules']):
+            condition = rule['condition']
+            action = rule['action']
+            config_methods = condition.get('methods')
+
+            if config_methods not in http_methods_lookup:
+                raise KeyError(
+                    f"invalid methods: {config_methods}"
+                    " should be one of 'read' or 'write'"
+                )
+            http_methods = http_methods_lookup[config_methods]
+
+            values = {
+                'listener_arn': f'${{{forward_lbl_tf_resource.arn}}}',
+                'action': {
+                    'type': 'forward',
+                    'forward': [],
                 },
-            },
-            'condition': [{
-                'http_request_method': {'values': ['POST', 'PUT', 'DELETE']},
-            }],
-            'depends_on': self.get_dependencies([forward_lbl_tf_resource]),
-        }
-        write_paths = resource.get('paths', {}).get('write')
-        if write_paths:
-            values['condition'].append(
-                {'path_pattern': {'values': write_paths}})
-        lblr_write_identifier = f'{identifier}-write'
-        lblr_write_tf_resource = \
-            aws_lb_listener_rule(lblr_write_identifier, **values)
-        tf_resources.append(lblr_write_tf_resource)
+                'condition': [{
+                    'http_request_method': {'values': http_methods},
+                    'path_pattern': {'values': [condition['path']]}
+                }],
+                'depends_on': self.get_dependencies([forward_lbl_tf_resource]),
+            }
+
+            weight_sum = 0
+            for a in action:
+                target_name = a['target']
+                if target_name not in valid_targets:
+                    raise KeyError(
+                        f'{target_name} not a valid target name'
+                    )
+
+                target_resource = valid_targets[target_name]
+
+                values['action']['forward'].append({
+                    'target_group': {
+                        'arn': f'${{{target_resource.arn}}}',
+                        'weight': a['weight']
+                    }
+                })
```

Comment:
according to the terraform docs and how previous code worked, `forward` here is a dict and the list of target groups arn/weight is actually in `target_group`

Reply:
@jfchevrette  this is what I have from the docs

```
# Weighted Forward action

resource "aws_lb_listener_rule" "host_based_routing" {
  listener_arn = aws_lb_listener.front_end.arn
  priority     = 99

  action {
    type = "forward"
    forward {
      target_group {
        arn    = aws_lb_target_group.main.arn
        weight = 80
      }

      target_group {
        arn    = aws_lb_target_group.canary.arn
        weight = 20
      }

      stickiness {
        enabled  = true
        duration = 600
      }
    }
  }

  condition {
    host_header {
      values = ["my-service.*.terraform.io"]
    }
  }
}
```

I had assumed that the list has each entry as `target_group` as the dict key. I have fixed it so now the list is inside `target_group`